---
author-meta:
- "Kim-Anh L\xEA Cao"
- Aedin C Culhane
- Elana Fertig
- Vincent J. Carey
- Jane Roe
bibliography:
- content/manual-references.json
date-meta: '2020-07-16'
header-includes: "<!--\nManubot generated metadata rendered from header-includes-template.html.\nSuggest improvements at https://github.com/manubot/manubot/blob/master/manubot/process/header-includes-template.html\n-->\n<meta name=\"dc.format\" content=\"text/html\" />\n<meta name=\"dc.title\" content=\"Community-wide hackathons establish foundations for emerging single cell data integration\" />\n<meta name=\"citation_title\" content=\"Community-wide hackathons establish foundations for emerging single cell data integration\" />\n<meta property=\"og:title\" content=\"Community-wide hackathons establish foundations for emerging single cell data integration\" />\n<meta property=\"twitter:title\" content=\"Community-wide hackathons establish foundations for emerging single cell data integration\" />\n<meta name=\"dc.date\" content=\"2020-07-16\" />\n<meta name=\"citation_publication_date\" content=\"2020-07-16\" />\n<meta name=\"dc.language\" content=\"en-US\" />\n<meta name=\"citation_language\" content=\"en-US\" />\n<meta name=\"dc.relation.ispartof\" content=\"Manubot\" />\n<meta name=\"dc.publisher\" content=\"Manubot\" />\n<meta name=\"citation_journal_title\" content=\"Manubot\" />\n<meta name=\"citation_technical_report_institution\" content=\"Manubot\" />\n<meta name=\"citation_author\" content=\"Kim-Anh L\xEA Cao\" />\n<meta name=\"citation_author_institution\" content=\"Melbourne Integrative Genomics, School of Mathematics and Statistics, University of Melbourne, Australia\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0003-3923-1116\" />\n<meta name=\"twitter:creator\" content=\"@mixOmicsTeam\" />\n<meta name=\"citation_author\" content=\"Aedin C Culhane\" />\n<meta name=\"citation_author_institution\" content=\"Data Sciences, Dana-Farber Cancer Institute, Boston, MA, USA\" />\n<meta name=\"citation_author_institution\" content=\"Biostatsitics, Harvard TH Chan School of Public Health, Boston, MA, USA\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0002-1395-9734\" />\n<meta name=\"twitter:creator\" content=\"@AedinCulhane\" />\n<meta name=\"citation_author\" content=\"Elana Fertig\" />\n<meta name=\"citation_author_institution\" content=\"Department of Oncology, Sidney Kimmel Comprehensive Cancer Center, Johns Hopkins University School of Medicine, Baltimore, MD, USA\" />\n<meta name=\"citation_author_institution\" content=\"Department of Biomedical Engineering, Johns Hopkins University School of Medicine, Baltimore, MD, USA\" />\n<meta name=\"citation_author_institution\" content=\"Department of Applied Mathematics and Statistics, Johns Hopkins University Whiting School of Engineering, Baltimore, MD, USA\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0003-3204-342X\" />\n<meta name=\"twitter:creator\" content=\"@FertigLab\" />\n<meta name=\"citation_author\" content=\"Vincent J. Carey\" />\n<meta name=\"citation_author_institution\" content=\"Channing Division of Network Medicine, Brigham and Women&#39;s Hospital, Harvard Medical School\" />\n<meta name=\"citation_author_orcid\" content=\"0000-0003-4046-0063\" />\n<meta name=\"citation_author\" content=\"Jane Roe\" />\n<meta name=\"citation_author_institution\" content=\"Department of Something, University of Whatever\" />\n<meta name=\"citation_author_institution\" content=\"Department of Whatever, University of Something\" />\n<meta name=\"citation_author_orcid\" content=\"XXXX-XXXX-XXXX-XXXX\" />\n<meta name=\"twitter:creator\" content=\"@XXX\" />\n<link rel=\"canonical\" href=\"https://BIRSBiointegration.github.io/whitePaper/\" />\n<meta property=\"og:url\" content=\"https://BIRSBiointegration.github.io/whitePaper/\" />\n<meta property=\"twitter:url\" content=\"https://BIRSBiointegration.github.io/whitePaper/\" />\n<meta name=\"citation_fulltext_html_url\" content=\"https://BIRSBiointegration.github.io/whitePaper/\" />\n<meta name=\"citation_pdf_url\" content=\"https://BIRSBiointegration.github.io/whitePaper/manuscript.pdf\" />\n<link rel=\"alternate\" type=\"application/pdf\" href=\"https://BIRSBiointegration.github.io/whitePaper/manuscript.pdf\" />\n<link rel=\"alternate\" type=\"text/html\" href=\"https://BIRSBiointegration.github.io/whitePaper/v/bc245d06d65f0997afd40b0fbb885aa67b77503f/\" />\n<meta name=\"manubot_html_url_versioned\" content=\"https://BIRSBiointegration.github.io/whitePaper/v/bc245d06d65f0997afd40b0fbb885aa67b77503f/\" />\n<meta name=\"manubot_pdf_url_versioned\" content=\"https://BIRSBiointegration.github.io/whitePaper/v/bc245d06d65f0997afd40b0fbb885aa67b77503f/manuscript.pdf\" />\n<meta property=\"og:type\" content=\"article\" />\n<meta property=\"twitter:card\" content=\"summary_large_image\" />\n<link rel=\"icon\" type=\"image/png\" sizes=\"192x192\" href=\"https://manubot.org/favicon-192x192.png\" />\n<link rel=\"mask-icon\" href=\"https://manubot.org/safari-pinned-tab.svg\" color=\"#ad1457\" />\n<meta name=\"theme-color\" content=\"#ad1457\" />\n<!-- end Manubot generated metadata -->"
keywords:
- markdown
- publishing
- manubot
lang: en-US
manubot-clear-requests-cache: false
manubot-output-bibliography: output/references.json
manubot-output-citekeys: output/citations.tsv
manubot-requests-cache-path: ci/cache/requests-cache
title: Community-wide hackathons establish foundations for emerging single cell data integration
...






<small><em>
This manuscript
([permalink](https://BIRSBiointegration.github.io/whitePaper/v/bc245d06d65f0997afd40b0fbb885aa67b77503f/))
was automatically generated
from [BIRSBiointegration/whitePaper@bc245d0](https://github.com/BIRSBiointegration/whitePaper/tree/bc245d06d65f0997afd40b0fbb885aa67b77503f)
on July 16, 2020.
</em></small>

## Authors



+ **Kim-Anh Lê Cao**<br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0003-3923-1116](https://orcid.org/0000-0003-3923-1116)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [mixOmicsTeam](https://github.com/mixOmicsTeam)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [mixOmicsTeam](https://twitter.com/mixOmicsTeam)<br>
  <small>
     Melbourne Integrative Genomics, School of Mathematics and Statistics, University of Melbourne, Australia
     · Funded by Grant National Health and Medical Research Council  Career Development fellowship (GNT1159458)
  </small>

+ **Aedin C Culhane**<br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-1395-9734](https://orcid.org/0000-0002-1395-9734)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [aedin](https://github.com/aedin)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [AedinCulhane](https://twitter.com/AedinCulhane)<br>
  <small>
     Data Sciences, Dana-Farber Cancer Institute, Boston, MA, USA; Biostatsitics, Harvard TH Chan School of Public Health, Boston, MA, USA
     · Funded by Chan Zuckerberg Initative, NIH, DoD (need to get grant IDs)
  </small>

+ **Elana Fertig**<br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0003-3204-342X](https://orcid.org/0000-0003-3204-342X)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [ejfertig](https://github.com/ejfertig)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [FertigLab](https://twitter.com/FertigLab)<br>
  <small>
     Department of Oncology, Sidney Kimmel Comprehensive Cancer Center, Johns Hopkins University School of Medicine, Baltimore, MD, USA; Department of Biomedical Engineering, Johns Hopkins University School of Medicine, Baltimore, MD, USA; Department of Applied Mathematics and Statistics, Johns Hopkins University Whiting School of Engineering, Baltimore, MD, USA
     · Funded by National Institute of Health, National Cancer Institute; National Institute of Health, National Institute of Dental and Craniofacial Research; Lustgarten Foundation; Emerson Foundation; Allegheny Health Network
  </small>

+ **Vincent J. Carey**<br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0003-4046-0063](https://orcid.org/0000-0003-4046-0063)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [vjcitn](https://github.com/vjcitn)<br>
  <small>
     Channing Division of Network Medicine, Brigham and Women's Hospital, Harvard Medical School
     · Funded by National Institutes of Health, National Human Genome Research Institute; National Institutes of Health, National Cancer Institute; Chan-Zuckerberg Initiative
  </small>

+ **Jane Roe**<br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [XXXX-XXXX-XXXX-XXXX](https://orcid.org/XXXX-XXXX-XXXX-XXXX)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [janeroe (PLEASE COPY/PASTE DO NOT EDIT THIS ONE)](https://github.com/janeroe (PLEASE COPY/PASTE DO NOT EDIT THIS ONE))
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [XXX](https://twitter.com/XXX)<br>
  <small>
     Department of Something, University of Whatever; Department of Whatever, University of Something
     · Funded by XX
  </small>



# Draft outline

<img src="images/Outline.png" alt="Outline of the main themes following our brainstorming sessions from the hackathons (KA Le Cao)" height="600"/>

Figure caption: Main challenges discussed during our brainstorming sessions from the hackathons


## Introduction [Figure 1 outline]

### Comprehensive characterization of biological systems with multi-omics
 - Single cell community has advanced technologies to enable concurrent processing of biological systems at multiple molecular resolutions
 - The lack of prior knowledge and gold standard benchmark naturally leads to a data-driven approach

### New single cell multi omics initiatives:
 - Human Cell Atlas (HCA): assess variation in normal tissues
 - Brain initiative and Allen Brain
 - Human Tumor Atlas Network (HTAN): Single-cell, longitudinal, and clinical outcomes atlases of cancer transitions for diverse tumor types. 

### What bulk multi-omics (e.g. TCGA, ENCODE) have taught us:
 - Type of omics that can answer a specific biological question
 - The value of open resources for methodological developments
 - New hypotheses

### Using hackathons to illustrate analysis standards and challenges for capturing biological information from multi-omics technologies 
 - Brief overview of our three hackathon studies highlighting state of the art challenges (e.g., spatial transcriptomics, cross-study analysis, epigenetic regulation)
 - Challenges include issues with noise and experimental design, Time lag between regulatory levels not addressed and many open questions
remain (e.g methylation / gene expression), Direction of regulation not captured
 - We present our findings from hackathon case studies that helped us obtain benchmarks and define a common language for multi-omics



## Hackathon case studies

### scRNA-seq + seqFISH as a case study for spatial transcriptomics [Figure 2 results]
#### Overview and biological question
#### Main challenges and how it was overcome by the participants
- Challenge 1: overlay of scRNA-seq onto seqFISH for resolution enhancement 
- Challenge 2: determine signatures of cellular co-localization or spatial coordinates in non-spatial scRNA-seq 


### Spatial proteomics and cross-study analysis [Figure 3 results]
#### Overview and biological question
#### Main challenges and how it was overcome by the participants
- Challenge 1: address the lack of overlap between proteins across studies
- Challenge 2: spatial protein expression analysis

### scNMT-seq as a case-study for epigenetic regulation [Figure 4 results], lead: Ricard Argelaguet
#### Overview and biological question
#### Main challenges and how it was overcome by the participants
- Challenge 1: defining genomic features
- Challenge 2: DNA methylation imputation
- Challenge 3: Linking epigenetic features / chromatin accessibility to gene expression


## Analytical approaches for hackathons and commonalities for multi-omics analysis illustrated by the series of hackathons [Figure 5 + Table 1]

Short introduction explaining that we focus here on the common challenges across hackathons

### Summary of hackathon study-specific methods
 - Table describes method, foundation in the context of previous bulk and single cell literature, and technology dependence
    - Attempts to tweak existing methods and challenges associated in hackathons
    - List methods that are either technology dependent (e.g. spatial) vs universal and how to choose them
  

### Dependence on pre-processing method and/or variable selection
- These steps are key and affect downstream analyses
  - Normalization / data transformation (seqFish), pre-processing, gene summaries (scNMT-seq) to variable selection (seqFish)
  - reproducibility difficult / no consensus. e.g. Alexis selected 19 genes whereas Zhu original paper based on 47 genes (difference in methods / processed data)
- Hackathon data pre-processed enable better comparisons across methods
- No consensus reached as those are emerging data with no ground truth nor established biological results

### Approaches for partial overlap of information (cells / features) and how to predict (cell type, dataset) using another data set
- Overlap in each study 
  - seqFish: same features but not cells; scProt: same proteins, not cells but similar patients; scNMT-seq: same cells but not features
  - How it was solved (Table)
- Anchoring information across datasets or studies is needed (Figure)
- Incorporation of existing biological knowledge
  - ‘From discovery to detection’ (Meuleman + debrief), time is ripe to include more knowledge in our data driven approaches
- Challenge: Partial cell overlap (but no features matching) and No overlap were not addressed

### Managing differences in scale and size for datasets that do not match cells or features 
- Hackathons datasets did not match cells or features.
  - scNMTQ-seq: MOFA limitation when # features vary (and size of datasets).
  - seqFish: greedy approach to select the best gene subset (Alexis, size); consider batch effect removal method (Amrit, scale)
 
- Consensus on projection based methods, even if pre-processing was applied (Table)
- Additional weighting is needed (e.g. Arora, Abadi).


## Interpretation challenges [Table 2: Glossary of terms]

### Interpretation requires a good understanding of the methods 
- Glossary needed for communication
- Incorporation of contiguous information to facilitate interpretation of analytical results 
- Biological knowledge and incorporation of information from databases are important, including bridges to data bases (KEGG, Gene Ontology, HCA) to validate through complementary data.

### Visualization tools for interpretation and communication
- Example: tSNE/ UMAP.

### Explaining results to biologists through generative models and simulations
- Example: factor analysis.

### Issues of over-discretization (premature-summarization) and over-simplification
- Example: notion of cell-type is insufficient (rare cell types vs. more “continuous” view on cell types).
- Problem with loss of information in the desire to simplify.
- Over interpretation / over reliance of graphical outputs


## Benchmarking in gold standard datasets [Figure 6, Table 3]

### Definition of benchmarking
- Goal 1: recovery of known cell types (processing of raw data, quantification, and clustering) 
- Goal 2: benchmark methods for their ability to discover known relationships between data modalities, e.g. gene regulatory relationships observed between chromatin accessibility and gene expression (relationships are not fully known at the single cell level).

### Strategies for benchmarking
- Simulation useful for known truth, but difficult to simulate realistic covariance structure across features and across data modalities.
- Benchmarking datasets for single cell studies (Table): limited focus on sequencing depth and diversity of cell types derived from a single assay:
ground truth for the intended effect of exposure in a proposed study design.
validation for a data integration task for which a new computational method may be proposed.
- Multi-modal assays benchmarking: 
- Design to address biological question (co-embedding, mapping or correlation, and inferring causal relationships). 
- Design for either data integration or further downstream analyses (e.g. differential analysis)
- Study design should take into account:
  - Biological and technical variability via replicates, block design, and randomization.
  - Power analysis for the intended effect or data integration task.
  - Dependencies between modalities.
- Challenge: No universal benchmark data scheme suits every combination of modality, benchmark datasets may be established for commonly used combinations of modalities or technologies, towards specific data integration tasks.

### Hackathons
- Cross-validation within study can be performed via Hackathons, e.g. cross-validation analysis of the scNMT-seq dataset using MOFA+
- Assess if relationships discovered in one dataset present in other datasets, potentially looking across single cell and bulk.
- Challenge: how to match dimensions of latent space across folds. (examples: permutation or cross-validation to assess model performance)

## Software infrastructure [Figure 7, Table 4]

### Key questions

- Q1: How should multimodal single cell data be managed for interactive and batch analyses?
- Q2: What methods will help software developers create scalable solutions for multimodal single cell analysis?
- Q3: How can we ensure that visualization methods that are central to multimodal single cell analysis are usable by researchers with visual impairments?

### Data management strategies

- Abstract data type with “multiassay experiment”: 
  - each mode =  different collection of features on possibly non-overlapping collections of samples;  
  - metadata on features conventionally defined 
  - metadata on samples include all relevant information on experimental conditions.
- Data container for a multi assay analysis:  
  - assays from the same cells, or measurements from distinct cells.
  - Assay slots containing variables or features from multiple modalities (e.g. gene expression units from scRNA-seq and protein units from sc-proteomics), feature may be multidimensional (e.g. spatial coordinates, locations of eQTLs).
- Map between the different assays to enable analysis

### Scalability strategies

- Reducing barriers to interpretable visualizations
- Color is a powerful data visualization tool to represent complex and rich scientific data. 
- Color vision deficiencies affect a substantial portion of the population. Include colorblind friendly visualizations [1] as a default setting in our visualizations


## Future directions

### What do we need for the future computation of multi omics single cell 

### Upcoming cell atlases technologies

- What is coming? 
- how they provide contexts for experimental perturbations
- How they provide context for novel datasets

### Unifying analysis goals for new computational methods

### Moving towards spatiotemporal omics and integration with mathematical models. 


## Figures, Tables and online resources
### Figures
- Figure 1: Outline of manuscript (Provided below)
- Figure 2: scRNA-seq + seqFISH main results
- Figure 3: scTargeted proteomics main results
- Figure 4: scNMT-seq main results
- Figure 5: Illustration of partial overlap between studies (at the cells and / or features level)
- Figure 6: Within study cross-validation
- Figure 7: Infrastructure of Giotto package as illustration

### Tables
- Table 1:  Summary of tasks and methods across all hackathons highlighting methods and common challenges (Provided below)
- Table 2: Glossary of terms, consensus terms will be used throughout the manuscript
- Table 3: Benchmarking single cell data sets
- Table 4: List of single cell analysis software

### Online resources
- Online resource 1: Three hackathon datasets (github)
- Online resource 2: R packages with open source reproducible vignettes (12 vignettes)


## Abstract {.page_break_before}


## Introduction


### Comprehensive characterization of biological systems with multi-omics
 - Single cell community has advanced technologies to enable concurrent processing of biological systems at multiple molecular resolutions
 - The lack of prior knowledge and gold standard benchmark naturally leads to a data-driven approach

### New single cell multi omics initiatives:
 - Human Cell Atlas (HCA): assess variation in normal tissues
 - Brain initiative and Allen Brain
 - Human Tumor Atlas Network (HTAN): Single-cell, longitudinal, and clinical outcomes atlases of cancer transitions for diverse tumor types. 

### What bulk multi-omics (e.g. TCGA, ENCODE) have taught us:
 - Type of omics that can answer a specific biological question
 - The value of open resources for methodological developments
 - New hypotheses

### Using hackathons to illustrate analysis standards and challenges for capturing biological information from multi-omics technologies 
 - Brief overview of our three hackathon studies highlighting state of the art challenges (e.g., spatial transcriptomics, cross-study analysis, epigenetic regulation)
 - Challenges include issues with noise and experimental design, Time lag between regulatory levels not addressed and many open questions
remain (e.g methylation / gene expression), Direction of regulation not captured
 - We present our findings from hackathon case studies that helped us obtain benchmarks and define a common language for multi-omics


- **Objectives of this paper**
    - Provide guidelines on tools / data / technologies / methods and needs to model the multi-scale regulatory processes in biological systems for a computational biologist audience


- **Outline and messages**
    - Cellular and molecular regulation is fundamentally multi-scale and captured by distinct data modalities
    - Traditional hypothesis-driven multi-omics/view studies only consider one facet of these technologies, but more can be learned through a holistic approach extending into atlases
    - We present our findings from hackathon case studies that helped us obtain a broader picture and language


    Outline of the paper:

<img src="images/Outline.png" alt="Outline of the main themes following our brainstorming sessions from the hackathons (KA Le Cao)" height="600"/>

Figure caption: Main challenges discussed during our brainstorming sessions from the hackathons


## scRNA-seq + FISH as a case study for spatial transcriptomics

### Overview and biological question  

Single cells are considered the smallest units and building blocks of each tissue, but they still require proper spatial and structural three-dimensional organization in order to assemble into a functional tissue that can exert its physiological function. In addition, most tissues are composed of multiple cell types whose identity and function can be inferred through their unique transcriptomic profile. In the last decade single-cell RNA-seq (scRNA-seq) played a key role to capture single cell gene expression profiles, which allowed us to map virtually all the different cell types and states in whole organisms. Despite this remarkable achievement this technology is based on cellular dissociation and hence does not maintain spatial relationships between single cells.  
More recently technologies have begun to emerge that can profile the transcriptome of single cells within their original environment. These technological advancements offer the possibility to examine how gene expression is influenced by cell-to-cell interactions or organized in a spatially coherent manner. One such approach is sequential single-molecule fluorescence in situ hybridization (seqFISH), which can identify single molecules at (sub)cellular resolution with high sensitivity. Nevertheless, and in contrast with scRNAseq, seqFISH and many other spatial transcriptomic technologies often pose significant technological challenges and hence the number of profiled genes per cell is usually restricted to a smaller number (10-100s).  
To overcome the lack of spatial information with scRNAseq and the common limited coverage in spatial datasets, we sought to combine and integrate a matching scRNAseq and seqFISH dataset, since both were generated from the mouse visual cortex region. More specifically, in this hackathon we explored a number of strategies to identify the most likely cell types in the seqfish dataset based on information obtained from the scRNAseq dataset. And in the opposite direction, we sought out how to transfer spatial information obtained from the seqfish dataset to that of the scRNAseq dataset.


### Computational challenges

#### Challenge 1: overlay of scRNA-seq onto seqFISH for resolution enhancement 

**[suggestion 1]**  
Sequencing and imaging based single-cell transcriptomic profiling have complementary strengths. Whereas single-cell RNAseq generates transcriptome-wide information, it does not have spatial information. On the other hand, seqFISH (Lubeck 2014; Shah 2016) provides single-cell resolution spatial information, but typically profiles the expression level of only 100-300 genes. Although the newer generation of seqFISH technology (called seqFISH+) has greatly enhanced its capacity which can now be used to profile 10,000 genes (Eng 2019), the technology is significantly more complex and costly. As such, it is desirable to develop computational approaches to effectively integrate scRNAseq and seqFISH data analyses.

In this hackathon, the participants were provided with seqFISH (Zhu 2018) and scRNAseq (Tasic 2016) data corresponding to the mouse visual cortex and challenged to accurately identify cell-types by integrating both datasets. Cell type labels, derived from scRNAseq analysis (Tasic 2016) and previous seqFISH/scRNAseq integration (Zhu 2018) were also provided as reference. A variety of computational approaches were applied to achieve this goal, including: supervised classification with support vector machines (Coulumb, Xu), semi-supervised self-training (Singh), and unsupervised matrix factorization methods (Sodicoff). While the methodologies are different, a number of themes recur, such as the importance of gene selection and batch effect correction. As expected, the ability to identify refined cell-type structure relies on the selection of cell-type specific marker genes in seqFISH data, suggesting a potential benefit of using single-cell RNAseq data to guide seqFISH experimental design. Batch effect is another important factor affecting the accuracy of data integration. While a number of batch effect correction methods have been developed (COMBAT, Seurat, Scanorama, etc), it remains challenging to distinguish technical from biological variations if the biological samples do not match exactly. 
 
**[suggestion 2]**  
The mouse visual cortex consists of multiple complex cell types, however the number of profiled genes in the seqFISH dataset is limited to 125 genes. Moreover, these genes were not prioritized based on their ability to discriminate between cell types and thus assigning the correct cell identity is challenging. In contrast the scRNAseq dataset is transcriptome wide, thus including the 125 aforementioned genes. As such we first used all genes to identify the cell type labels for each cell in the scRNAseq data with high certainty. Next, we leveraged that information to build a classifier based on (a subset of) the 125 common genes only. This classifier could subsequently be applied to the seqFISH dataset in order to assign cell types to each cell with high probability.

During the hackathon we tested various machine learning or data integration models, but also noted that initial dataset normalization strategies might have a significant impact on the final results (see common challenges @Amrith). Although unique molecular identifier (UMI) based scRNAseq and seqFISH can both be considered as count data, we observed dataset specific biases that could be attributed to either platform (imaging vs sequencing) or sample specific sources of variation. Here we opted to apply a quantile normalization approach which forces a similar expression distribution for each shared gene. 
The hackathon participants selected two machine learning classifiers, a supervised support vector machine (SVM) and a semi-supervised lasso and elastic-net regularized generalized linear model (glmnet). To further improve the SVM model the participants tried multiple kernels and searched for optimal hyperparameters using a combined randomized and zoomed search. In addition, different flavors of recursive feature elimination was used to find the optimal or minimum number of genes needed to correctly classify the majority of the cells. Importantly, participants tested different classification accuracy metrics to alleviate the major class imbalance in the dataset, since more than 90% of cells were excitatory or inhibitory neurons. The glmnet approach used a iterative model building approach, which combines both datasets and initially only retains the highest confidence labels and then gradually adds more cell type labels until all cells have been classified. This type of self training approach might in fact be more generalizable to other datasets. 
Finally, a data integration approach based on integrative non-negative matrix factorization (NMF) was used by applying the previously published LIGER approach. By integrating both datasets in a similar subspace based on shared factors, cell type labels can be transferred using a nearest neighbor approach.



#### Challenge 2: determine signatures of cellular co-localization or spatial coordinates in non-spatial scRNA-seq 

**[suggestion 1]**  
How could one identify spatial patterns in the seqFISH data? This broad question can be divided into a number of specific tasks, such as detecting genes whose expression is spatially coherent, cell types whose spatial distribution is confined to distinct regions, recurrent multi-cell-type interaction clusters, etc, and the spatial scale may vary from subcellular all the way to tissue-wide organizations. While there have been abundant studies in the geo-spatial analysis domain, computational tools targeting specific spatial transcriptomic questions are still lacking. In this hackathon, one group tackled this challenge by aggregating gene expression data from neighboring cells followed by spatial clustering (Coulumb). Much more work in the future is need to further explore such information.  

**[suggestion 2]**  
Most analyses that were originally developed for scRNAseq data can be immediately applied to spatial transcriptomic datasets, however methods to extract sources of variation that originate from spatial factors are still sparse. To incorporate spatial information the cells from the seqFISH dataset were first connected through a spatial network based on Voronoi tessellation and then the expression of each individual cell was spatially smoothed by calculating the average gene expression levels over all the neighboring cells. This smoothened and aggregated data matrix was subsequently used to create a 2 dimensional UMAP from which clusters were identified through a density based clustering approach. The obtained cluster labels can then be mapped back to the original spatial locations  for further visual inspection and analysis.




## Spatial proteomics and cross-study analysis

### Overview and biological question

### Computational challenges

#### Challenge 1: address the lack of overlap between proteins across studies
#### Challenge 2: spatial protein expression analysis


## scNMT-seq as a case-study for epigenetic regulation

### Overview and biological question

### Computational challenges

#### Challenge 1: defining genomic features
#### Challenge 2: DNA methylation imputation
#### Challenge 3: Linking epigenetic features / chromatin accessibility to gene expression


## Analytical approaches for hackathons and commonalities for multi-omics analysis illustrated by the series of hackathons [Figure 5 + Table 1]

Short introduction explaining that we focus here on the common challenges across hackathons

### Summary of hackathon study-specific methods
 - Table describes method, foundation in the context of previous bulk and single cell literature, and technology dependence
    - Attempts to tweak existing methods and challenges associated in hackathons
    - List methods that are either technology dependent (e.g. spatial) vs universal and how to choose them
  

### Dependence on pre-processing method and/or variable selection
- These steps are key and affect downstream analyses
  - Normalization / data transformation (seqFish), pre-processing, gene summaries (scNMT-seq) to variable selection (seqFish)
  - reproducibility difficult / no consensus. e.g. Alexis selected 19 genes whereas Zhu original paper based on 47 genes (difference in methods / processed data)
- Hackathon data pre-processed enable better comparisons across methods
- No consensus reached as those are emerging data with no ground truth nor established biological results

### Approaches for partial overlap of information (cells / features) and how to predict (cell type, dataset) using another data set
- Overlap in each study 
  - seqFish: same features but not cells; scProt: same proteins, not cells but similar patients; scNMT-seq: same cells but not features
  - How it was solved (Table)
- Anchoring information across datasets or studies is needed (Figure)
- Incorporation of existing biological knowledge
  - ‘From discovery to detection’ (Meuleman + debrief), time is ripe to include more knowledge in our data driven approaches
- Challenge: Partial cell overlap (but no features matching) and No overlap were not addressed

### Managing differences in scale and size for datasets that do not match cells or features 
- Hackathons datasets did not match cells or features.
  - scNMTQ-seq: MOFA limitation when # features vary (and size of datasets).
  - seqFish: greedy approach to select the best gene subset (Alexis, size); consider batch effect removal method (Amrit, scale)
 
- Consensus on projection based methods, even if pre-processing was applied (Table)
- Additional weighting is needed (e.g. Arora, Abadi).


<img src="images/summary_fig_person.png" alt="Common challenge: partial overlap between multi modal data sets (KA Le Cao)" height="400"/>
Figure caption:

A) scSpatial: required overlap of features (genes), but cells do not overlap. Cell type prediction for seqFISH data was performed based on scRNA-seq (known) [credit: Amrit Singh / Kris S]
B) scProteomics: a small number of features overlap (proteins) but patients and cells do not overlap. Data imputation (?), spatial co-localization or cell type prediction was performed   [credit: Lauren Hsu and Pratheepan Jenagan / Kris S]
C) scNMT-seq: cells are matching across assays but features do not overlap. Data integration was performed [credit: Al Abadi / Kris S]
D) Nor cells nor features are matching  [credit: Chen Meng / Kris S]


<style type="text/css">
.tg  {border-collapse:collapse;border-spacing:0;}
.tg td{border-color:black;border-style:solid;border-width:1px;font-family:Arial, sans-serif;font-size:14px;
  overflow:hidden;padding:10px 5px;word-break:normal;}
.tg th{border-color:black;border-style:solid;border-width:1px;font-family:Arial, sans-serif;font-size:14px;
  font-weight:normal;overflow:hidden;padding:10px 5px;word-break:normal;}
.tg .tg-c8qn{background-color:#ffffff;color:#333333;font-weight:bold;text-align:left;vertical-align:top}
.tg .tg-7euo{background-color:#ffffff;color:#333333;text-align:left;vertical-align:top}
</style>
<table class="tg">
<thead>
  <tr>
    <th class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Common challenges</span></th>
    <th class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Tasks</span></th>
    <th class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">seqFISH case study</span></th>
    <th class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Sc targeted proteomics</span></th>
    <th class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">scNMT-seq</span></th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Normalisation, transformations, pre-processing </span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Check data distribution (Coullomb, Singh)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">HVG (Xu)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Row and column wise, VSN (Meng)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Arcsinh transformation  (Jeganathan).</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Inverse transformation (Jenagan)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Selection of TN patients (Jenagan)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Summaries of DNA measurements (data provided in hackathon)</span><br></td>
  </tr>
  <tr>
    <td class="tg-c8qn" rowspan="4"><span style="font-weight:700;font-style:normal;text-decoration:none">Overlap</span></td>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Partial feature overlap </span><span style="font-weight:400;font-style:normal;text-decoration:none">(but cells not matching)</span></td>
    <td class="tg-7euo"></td>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Imputation</span><span style="font-weight:400;font-style:normal;text-decoration:none">:</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Direct inversion latent variable analysis for prediction (Sankaran)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Optimal transport to predict protein expression (Lin)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">KNN averaging (Jenathan*)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">No imputation:</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Biological Network Interaction (Foster*)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Removed</span><br></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Partial cell overlap </span><span style="font-weight:400;font-style:normal;text-decoration:none">(but no features matching)</span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">MBPCA (Meng*)</span></td>
    <td class="tg-7euo"></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">No cell overlap </span><span style="font-weight:400;font-style:normal;text-decoration:none">but some / all feature overlap</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Averaging nearest neighbors in latent space to impute unmeasured expression values (Coullomb?)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Topic modelling to predict cell spatial co-location or spatial expression (Jenathan), partial feature overlap</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Transfer cell type label with Random Forest (Hsu), all feature overlap?</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">LIGER - NMF (Welch), all feature overlap</span></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">No overlap </span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">RLQ (Chen*)</span></td>
    <td class="tg-7euo"></td>
  </tr>
  <tr>
    <td class="tg-c8qn" rowspan="2"><span style="font-weight:700;font-style:normal;text-decoration:none">Generic approaches </span><br><span style="font-weight:400;font-style:italic;text-decoration:none">Distinction btw ‘bulk’ =   adapted from bulk omics vs</span><br><span style="font-weight:400;font-style:italic;text-decoration:none">‘Sc’ = developed specifically for single cell</span></td>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Classification &amp; feature selection</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Backward selection with SVM (Coullomb)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">self training ENet (Singh)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Balanced error rate (Coullomb, Singh)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Recursive Feature Elimination (Xu)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">(all bulk)</span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Supervised clustering MOSAIC (Arora) (bulk)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Lasso in regression-type models (bulk)</span><br></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Cell type prediction</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Projections / clustering (Sodicoff*)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">SVM (Coullomb, Xu)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">ssEnet (Singh) </span><br><span style="font-weight:400;font-style:normal;text-decoration:none">(all bulk)</span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Study specific approaches</span><br><span style="font-weight:400;font-style:italic;text-decoration:none">Distinction btw ‘bulk’ =   adapted from bulk omics vs</span><br><span style="font-weight:400;font-style:italic;text-decoration:none">‘Sc’ = developed specifically for single cell</span></td>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Spatial analysis</span><br><br><br><br></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">HMRF</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Voronoi tesselation (Coullomb) (bulk)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Spatial autocorrelation (Moran index, Hsu, Lin)</span><br><br><span style="font-weight:400;font-style:normal;text-decoration:none">Select spatial discriminative features:</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Moran Index, NN correlation, Cell type</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">interaction composition, </span><br><span style="font-weight:400;font-style:normal;text-decoration:none">L function (Lin)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">(all bulk?)</span></td>
    <td class="tg-7euo"></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Managing differences in scale</span><br><span style="font-weight:400;font-style:italic;text-decoration:none">Distinction btw ‘bulk’ =   adapted from bulk omics vs</span><br><span style="font-weight:400;font-style:italic;text-decoration:none">‘Sc’ = developed specifically for single cell</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Data integration</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">LIGER - NMF (Sodicoff*) (sc)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">ComBat (Singh*)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Projection methods MFA, sGCCA  (Singh*)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">UMAP / tSNE (Sodicoff*) (sc)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Multi-block PCA</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Weighting matrices based on their similarities: Data wise (STATIS, MFA) (Chen*)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">(bulk)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Scale CyTOF and MIBI-TOF to the range of CyTOF values (Jenagan)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">LIGER - NMF (Welch) (bulk)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">Projection to Latent Structures (Abadi) (bulk)</span><br><span style="font-weight:400;font-style:normal;text-decoration:none">MOSAIC as weighted approach (Arora) (bulk)</span></td>
  </tr>
  <tr>
    <td class="tg-c8qn"><span style="font-weight:700;font-style:normal;text-decoration:none">Inclusion of additional information </span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Inclusion of additional information </span></td>
    <td class="tg-7euo"></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Survival prediction: Cox regression based on spatial features (Lin)</span></td>
    <td class="tg-7euo"><span style="font-weight:400;font-style:normal;text-decoration:none">Include annotated DHS Index to  anchor new/unseen data from DNase-seq, (sc)ATAC-seq, scNMT-seq, etc. for </span><span style="font-weight:400;font-style:italic;text-decoration:none">De novo</span><span style="font-weight:400;font-style:normal;text-decoration:none"> peak calling(Meuleman*) (bulk)</span></td>
  </tr>
</tbody>
</table>

Table: Different methods were used in the hackathon. {#tbl:example-id}


## Challenges for interpretation


### Interpretation for quantitative team: requires a good understanding of the methods (no black boxes, a common language)

Communicating within the field: what approaches are we talking about, this requires agreement on a glossary (ref: Table1). 
Supervised versus unsupervised methods.
Visualization figures are useful for the mathematical and data science team in the Explorator phase. 

### Interpretation for biologists:
Understanding the output from the analyses of the data is facilitated by the incorporation of contiguous information.
Redundant biological knowledge and incorporation of information from databases are important in the workflow.
Biological interpretations are facilitated by bridges to databases such as KEGG, Gene Ontology, Human Cell Atlas, Biomart and many other databases.
Validation  through complementary data and sequential experimental design.

### Visualization tools for interpretation and communication to biologists
There are pitfalls in using sophisticated graphics which lead to over-interpretation or misinterpretation
(size of clusters in tSNE related to sampling baselines rather than density, ...)
Example of effective visual interpretation tools : brushing UMAP (Kris Sankaran).

### Explaining results to biologists through generative models and simulations (ex: Factor Analysis).
Several difficulties arise when explaining summaries and conclusions, problems encountered include non-identifiability of models
or non-sufficiency of summaries, simulations can often provide effective communication tools.


### Issues of over-discretization, over-simplification
Example 1: The notion of cell-type is insufficient (Communication challenge with biologists about tradeoffs between focusing on rare cell types vs. more “continuous” view on cell types).

Problem with loss of information in the desire to simplify.


### Counterexamples


## Techniques and challenges for benchmarking methods

### We must first define what we are benchmarking

 * Often the goal in benchmarking is recovery of known cell types with
   processing of raw data, quantification, and clustering. The
   Adjusted Rand Index (ARI) or other metrics for partitions are
   used.
 * One may also attempt to benchmark methods for their ability to
   discover known relationships between data modalities, e.g. gene
   regulatory relationships observed between chromatin accessibility
   and gene expression. However, this is made difficult by the fact
   that these relationships are not fully known at the single cell
   level.

### Strategies for benchmarking

* Simulation is useful for having known truth, but it is difficult to
  simulate realistic covariance structure across features and across
  data modalities.
* Benchmarking datasets (add examples from Google Doc). Benchmark
  datasets for single cell studies have largely centered around
  measuring sequencing depth and diversity of cell types derived from
  a single assay of interest (e.g. scRNAseq). A benchmark dataset
  serves a few purposes:
    - Provides ground truth for the intended effect of exposure in a
      proposed study design.
    - Provides validation for a data integration task for which a new
      computational method may be proposed.

  For multi-modal assays, while the intended effects can vary based on
  the leading biological questions, one may abstract out common data
  integration tasks such as co-embedding, mapping or correlation, and
  inferring causal relationships. We distinguish data integration from
  further downstream analyses that may occur on integrated samples
  such as differential analysis of both assays with regard to to a certain
  exposure.

  Both the intended effects and data integration task rely on study
  design that takes into account:
    - Biological and technical variability via replicates, block
      design, and randomization.
	- Power analysis for the intended effect or data integration
      task.
	- Dependencies between modalities, for e.g. gene expression
      depending on gene regulatory element activity, requires that
      experiment design must also account for spatial and temporal
      elements in sampling for a given observation.

  As such, no universal benchmark data scheme may suit every
  combination of modality, and benchmark datasets may be established
  for commonly used combinations of modalities or technologies,
  towards specific data integration tasks.

* Cross-validation within study can be performed. For example the
  following 
  [cross-validation analysis of the scNMT-seq dataset](https://github.com/Wancen/CV-MOFA)
  was performed using MOFA+

  <img src="images/scNMT_MOFA_CV.png" alt="Cross validation of MOFA+ on scNMT-seq" height="300"/>

  A challenge with within study cross-validation is how to
  match dimensions of latent space across folds. (add examples from
  Google Doc of papers that have performed either permutation or
  cross-validation to assess model performance)
* Cross-study validation would assess if relationships discovered in
  one dataset present in other datasets, potentially looking across
  single cell and bulk.


## Software strategies to enable analyses of multimodal single cell experiments

### Key questions

* How should multimodal single cell data be managed for interactive and batch analyses?
* What methods will help software developers create scalable solutions for multimodal single cell analysis?
* How can we ensure that visualization methods that are central to multimodal single cell analysis
are usable by researchers with visual impairments?

### Data management strategies

* Abstract data type: "multiassay experiment".  This reflects the idea that each mode will
be characterized by a different collection of features on possibly non-overlapping collections
of samples.  The metadata on features should be clearly and conventionally defined.  For example,
genes and transcripts are enumerated using Ensembl catalog identifiers; regions of accessibility
are defined using genomic coordinates in a clearly specified reference build.  Metadata on
samples must include all relevant information on experimental conditions such as treatment,
protocol, and date of technical processing.

(More fodder-AS)

**Key points:**
1) What do we want to store and share? Data object vs analysis object. How would the design change based on what is stored?
2) Do we need a flexible, universal framework (e.g. MAE) or an experiment class for every possible combination of modalities or technologies?
3) Do we have adequate data representation for all “assays”?

   - Multi-modal single cell data may consist of multi-assay measurements from the same cell (e.g. CITE-seq, sci-CAR) or integration of multi-assay measurements from distinct cells from the same or distinct starting samples. A sample here refers to the biological specimen of origin (tissue A from individual X).
A data container for a multi assay analysis must hold
   1) Assay slots containing variables or features from multiple modalities (e.g. gene expression units from scRNA-seq and protein units from sc-proteomics). In some cases, the feature may be multidimensional (e.g. spatial coordinates, locations of eQTLs).
   2) Observations or cell identities
   3) Metadata for sample of origin for the individual cells, e.g. study, center, phenotype, perturbation.
   4) A map between the different assays to enable analysis

   - The MAE is such a Bioconductor container for overlapping observations, and may serve as a starting point for further expansion. Besides the primary data elements for storing “data objects”, the summarizedExperiment class offers attributes and Methods for storing results of analysis, as an “analysis object”.
      - While common assays such as RNA-seq and ATAC-seq have well-defined data representations (e.g. transcript names), data representation need to be defined newer assays, which may need multiple dimensions for adequate definition (e.g. x, y, z coordinates for images).
      - The observations of different modalities may not be directly comparable (e.g. RNA may be measured from individual cells but spatial transcriptomics may cover a few cells in the matched area).  
      - In the absence of universal standards, the metadata may vary from analysis to analysis.
      - It is crucial that data containers use consistent assay access methods (possibly through methods inheritance. e.g. from `SummarizedExperiment`). This will ensure less redundancy in development process and allow powerful implementation strategies.


(Note, standard BioC container/class terms may not be correctly used. End of fodder - AS)

* Serializations and data access methods for
    * spatial transcriptomics
    * scNMT-seq ...


### Scalability strategies

#### An overview of our suggested approach for compilation of the hackathon contributions
The contributed challenge analyses as well as the used datasets may be collated into software packages which include their specific dependencies. In such a setting, the analysis packages can simply use the corresponding dataset package(s) as a dependency. Additionally, if the challenge datasets have undergone preprocessing it is essential that all the preprocessing steps are outlined and made available in the dataset packages. This will facilitate transparency and analysis reproducibility, as well as allow Continuous Integration (CI) of the analyses and preprocessing changes and Continuous Delivery (CD) of the analysis reports. The CI/CD workflow may also be automated on a hosted server and containerized reports can be generated for enhanced efficiency and portability, respectively.

### Reducing barriers to interpretable visualizations
Color is a powerful data visualization tool that helps representing the different dimensions of our increasingly complex and rich scientific data.
Color vision deficiencies affect a substantial portion of the population.
Therefore,  it is desirable to aim towards presenting scientific information in a manner that is as accessible as possible for all readers. 
Color vision deficiency leads to difficulties in perceiving patterns (the basis for the Ishihara's color vision tests) in multi-colored figures.
In rare cases, the perceived patterns; e.g. in heatmaps and reduced dimension plots, can differ between individuals with normal and color deficient vision.

One strategy to address these issues is to include colorblind friendly visualizations [@https://doi.org/10.1038/nmeth.1618] as a default setting in our visualizations.
Several colorblind-friendly palettes exist (e.g., see R packages [viridis](https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html) and [dittoSeq](https://github.com/dtm2451/dittoSeq)) and can be integrated into data presentation as the default option.
Even with these palettes in place, it is desirable to limit the number (about 8-10 at a maximum) of colors in visualizations.
To reduce the dependence on colors, one solution would be to include additional visual cues to differentiate regions (hatched areas) or cells (point shapes).
Overall, a broader discussion regarding the accessibility of our figures that is not just limited to color vision deficiencies would be greatly beneficial towards improving data accessibility.
Perhaps one tool to address broader accessibility could be the inclusion an "accessibility caption" accompanying figures which "guide" the reader's perception of the images.


[Reference 1: Color Coding](https://www.nature.com/articles/nmeth0810-573)

[Reference 2: Points of View: Color Blindness](https://www.nature.com/articles/nmeth.1618)

[Viridis Color Palettes](https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html)

[An overview of the issues with impaired color perception](https://www.vwvj.be/sites/default/files/zien/zien_-_uit_voorbije_vorming/kleurzinonderzoekextratips.pdf)

[US Government tools for accessibility](https://accessibility.18f.gov/tools/)

### Details of working components -- trimmed

you can interact with underlying data at [google sheet](https://docs.google.com/spreadsheets/d/1tSUQ9iDKqq72TB9G3Cx1evg2sekS-ytx5wRXDv6vxfg/edit?usp=sharing)

|Type|Brief name (link)|Description|
|----|-----------------|-----------|
|Matlab package|[CytoMAP](https://gitlab.com/gernerlab/cytomap)|CytoMAP: A Spatial Analysis Toolbox Reveals Features of Myeloid Cell Organization in Lymphoid Tissues|
|Matlab package|[histoCAT](https://github.com/BodenmillerGroup/histoCAT)|histoCAT: analysis of cell phenotypes and interactions in multiplex image cytometry data|
|Python library|[PyTorch](https://pytorch.org)|General framework for deep learning|
|Python package|[SpaCell](https://github.com/BiomedicalMachineLearning/SpaCell)|SpaCell: integrating tissue morphology and spatial gene expression to predict disease cells|
|Python package|[Scanpy](https://github.com/theislab/scanpy)|Python package for single cell analysis|
|R data class|[MultiAssayExperiment](https://bioconductor.org/packages/MultiAssayExperiment)|unify multiple experiments|
|R data class|[SpatialExperiment](https://github.com/drighelli/SpatialExperiment)|SpatialExperiment: a collection of S4 classes for Spatial Data|
|R package|[Giotto](https://github.com/RubD/Giotto)|Spatial transcriptomics|
|R package|[cytomapper](https://github.com/BodenmillerGroup/cytomapper)|cytomapper: Visualization of highly multiplexed imaging cytometry data in R|
|R package|[Spaniel](https://github.com/RachelQueen1/Spaniel/)|Spaniel: analysis and interactive sharing of Spatial Transcriptomics data|
|R package|[Seurat](https://github.com/satijalab/seurat)|R toolkit for single cell genomics|
|R package|[SpatialLIBD](https://github.com/LieberInstitute/spatialLIBD)|Transcriptome-scale spatial gene expression in the human dorsolateral prefrontal cortex|
|R package|[Cardinal](https://cardinalmsi.org/)|Cardinal: an R package for statistical analysis of mass spectrometry-based imaging experiments|
|R package|[CoGAPS](https://github.com/FertigLab/CoGAPS)|scCoGAPS learns biologically meaningful latent spaces from sparse scRNA-Seq data|
|R package|[projectR](https://github.com/genesofeve/projectR)|ProjectR is a transfer learning framework to rapidly explore latent spaces across independent datasets|
|R package|[SingleCellMultiModal](https://github.com/waldronlab/SingleCellMultiModal)|Serves multiple datasets obtained from GEO and other sources and represents them as MultiAssayExperiment objects|
|R scripts|[SpatialAnalysis](https://github.com/drighelli/SpatialAnalysis)|Scripts for SpatialExperiment usage|
|Self-contained GUI|[ST viewer](https://github.com/jfnavarro/st_viewer)|ST viewer: a tool for analysis and visualization of spatial transcriptomics datasets|
|Shiny app|[Dynverse](https://zouter.shinyapps.io/server/)|A comparison of single-cell trajectory inference methods: towards more accurate and robust tools|
|R package|[mixOmics](https://github.com/mixOmicsTeam/mixOmics)|R toolkit for multivariate analysis of multi-modal data|
|Python package|[totalVI](https://github.com/YosefLab/scVI)|A variational autoencoder (deep learning model) to integrate RNA and protein data from CITE-seq experiments|
|Python web application||[ImJoy](https://imjoy.io/#/)|Deep learning for image analysis|
|Python package|[napari](https://github.com/napari/napari)|Interactive big multi-dimensional 3D image viewer|
|Software|[QuPath](https://qupath.github.io/)|Multiplex whole slide image analysis|
|Python package|[Cytokit](https://github.com/hammerlab/cytokit)|Multiplex whole slide image analysis|
|Python package|[cmIF](https://gitlab.com/engje/cmif)|Multiplex whole slide image analysis|
|Software|[Facetto](https://github.com/kruegert/facetto)|Multiplex whole slide image analysis, not available yet|
|Software, Python based|[CellProfiler](https://cellprofiler.org/)|Image analysis|

Here is the schematic of SpatialExperiment class from Dario Righelli.

<img src="images/SpatialExperimentScheme1.1.png" alt="SpatialExperiment (D. Righelli)" height="400"/>


Here is the schematic of how seqFISH data are stored in the SingleCellMultiModal package from Dario Righelli.

<img src="images/scMultiModal_seqFISH.png" alt="seqFISH (D. Righelli)" height="400"/>



## Discussion

### Emerging analytical methods and technologies

### Community needs for data structures, analysis methods, etc


## Glossary

| Consensus Term | Related Terms | Description | Citation |
|----------|----------|----------|----------|
| network | graph, adjacency matrix | A set of *nodes*, representing objects of interest, linked by *edges*, representing specific relationships between nodes. | ? |
| node | vertex | Element of interest in a network and linked to other nodes. For example: people, cells, proteins or genes. Nodes can have several properties called *attributes* like cell type or position. | ? |
| edge | link | The relationship between 2 nodes in a network. For example: friendship in social networks, cells in contact in a spatial network, or gene-gene interactions in a gene regulatory network. | ? |
| concordant | concordant, coherent, consistent | ? | @doi:10/gghpmf |
| contributions | variable weights, loadings, eigenvector, axis, direction, dimension, coefficients, slopes | Contributions of the original variables in constructing the components. | @doi:10/gcj84s, @doi:10/f52wrr |
| latent factors | variates, scores, projections, components, latent/hidden/unobserved variables/factors | Weighted linear combinations of the original variables. | @doi:10/gcj84s, @doi:10/f52wrr |
| multimodal | Multiview, multiway arrays, multimodal, multidomain, multiblock, multitable, multi-omics, multi-source data analysis methods, N-integration | Methods pertaining to the analysis of multiple data matrices for the same set of observations. | @doi:10/gcj84s, @doi:10/gf8dqn, @doi:10/f83qvd |
| conjoint analysis | conjoint analysis, P-integration, meta-analysis, multigroup data analysis | Methods pertaining to the analysis of multiple data matrices for the same set of variables. | @doi:10/gcj84s, @doi:10/f52wrr, @doi:10/f7nnbm |
| variable | feature, variable | A measurable quantity that describes an observation's attributes. Variables from different modalities include age, sex, gene or protein abundance, single nucleotide variants, operational taxonomic units, pixel intensity *etc.* | ? |
| biomarker | marker, biomarker | A variable that is associated with normal or disease processes, or responses to exposures, or interventions. Any change in this variable is also associated with a change in the associated clinical outcome. These variables may be used for diagnostic, monitoring, Pharmacodynamic responses. Examples include LDL cholesterol, CD4 counts, hemoglobin A1C. | @doi:10/gcxh8n |
| panel | biomarker panel, biomarker signature | A subset of the originally measured variables that are determined to be associated with the outcome or response variable. This may be determined using statistical inference, feature selection methods, or machine/statistical learning.  | @doi:10/d2cm, @doi:10/d2cn |
| observation | sample, observation, array | A single entity belonging to a larger grouping. Examples include patients, subjects, participants, cells, biological sample, usually the unit of observation on which the variables are measured *etc.* | ? |

Table: Glossary of interchangeable terms in the field of single-cell and bulk multi-omics (multi-source) data analysis. {#tbl:table_gloss}


## References {.page_break_before}

<!-- Explicitly insert bibliography here -->
<div id="refs"></div>
