## Introduction


### Comprehensive characterization of biological systems with multi-omics
 - Single cell community has advanced technologies to enable concurrent processing of biological systems at multiple molecular resolutions
 - The lack of prior knowledge and gold standard benchmark naturally leads to a data-driven approach

### New single cell multi omics initiatives:
 - Human Cell Atlas (HCA): assess variation in normal tissues
 - Brain initiative and Allen Brain
 - Human Tumor Atlas Network (HTAN): Single-cell, longitudinal, and clinical outcomes atlases of cancer transitions for diverse tumor types.

### What bulk multi-omics (e.g. TCGA, ENCODE) have taught us:
 - Type of omics that can answer a specific biological question
 - The value of open resources for methodological developments
 - New hypotheses

### Using hackathons to illustrate analysis standards and challenges for capturing biological information from multi-omics technologies
 - Brief overview of our three hackathon studies highlighting state of the art challenges (e.g., spatial transcriptomics, cross-study analysis, epigenetic regulation)
 - Challenges include issues with noise and experimental design, Time lag between regulatory levels not addressed and many open questions
remain (e.g methylation / gene expression), Direction of regulation not captured
 - We present our findings from hackathon case studies that helped us obtain benchmarks and define a common language for multi-omics


- **Objectives of this paper**
    - Provide guidelines on tools / data / technologies / methods and needs to model the multi-scale regulatory processes in biological systems for a computational biologist audience


- **Outline and messages**
    - Cellular and molecular regulation is fundamentally multi-scale and captured by distinct data modalities
    - Traditional hypothesis-driven multi-omics/view studies only consider one facet of these technologies, but more can be learned through a holistic approach extending into atlases
    - We present our findings from hackathon case studies that helped us obtain a broader picture and language


    Outline of the paper:

![Main challenges discussed during our brainstorming sessions from the hackathons.](images/Outline.png){#fig:outline-again}

## Overview of hackathon studies to illustrate multi-omics standards and challenges

Multi-omics technologies provide the unique capacity for full spatial and molecular characterization biological systems. In these analyses, each data modality can uniquely resolve specific biological scales from which complementary data integration techniques can resolve multi-scale interactions. In spite of this potential, computational techniques and benchmark strategies for even pairs of datasets across high-throughput measurement technologies remain an active area of research. The hackathon studies at the Mathematical Frameworks for Integrative Analysis of Emerging Biological Data Workshop and described here were tailored to highlight independent challenges of data integration. First, while spatial molecular profiling is rapidly emerging, these technologies often provide lower molecular resolution than non-spatial counterparts. In these cases, integration strategies that merge these datasets have the promise to enhance the molecular resolution of spatially resolved profiling. To address this challenge, We designed a hackathon using spatially resolved transcriptional data from seqFISH with corresponding non-spatial single cell profiling data for the mouse visual cortex from [@doi:10.1038/nn.4216, @doi:10.1038/nbt.4260]. A second challenge for multi-omics technologies in cases for which tissue availability limits the ability to obtain multiple measurements in samples from identical conditions. This limitation raises the question as to whether information can be transferred from datasets between disparate sample cohorts. Therefore, we designed the second hackathon to contain two triple negative breast cancer cohorts profiled with single cell proteomics profiling from mass cytometry (CyTOF) [@doi:10.1016/j.cell.2019.03.005] and spatial in-situ proteomics from Multiplexed Ion Beam Imaging (MIBI) [@doi:10.1016/j.cell.2018.08.039]. While each of these hackathons use data at the same molecular scale, genetic and epigenetic alterations to DNA further drive the transcriptional regulation that mediates intra- and inter-cellular signaling processes underlying cellular fate transitions and states. Our third hackathon uses scNMT-seq data to obtain concurrent DNA methylation, chromatin accessibility, and RNA expression from the same cells to delineate the regulatory networks that underlie mouse gastrulation [@doi:10.1093/humupd/dmy021]. Altogether, exploration of the analysis approaches employed to address these disparate hackathons across biological contexts provides an unique opportunity to identify technology-specific challenges and unifying themes that are essential to effectively employ multi-omics datasets into new biological knowledge. 
